src/inversion_scripts/plumes.py

```diff
@@ -0,0 +1,753 @@
+            except:
+                msg = (
+                    f'Cannot open cached file at {infile}, fetching new data.'
+                )
+                warnings.warn(msg)
+                self.gdf = self._get_data()
+        
+        else:
+            self.gdf = self._get_data()
+        
+        # cache all data
+        self._cache_data()
+        
+        # filter data according to 
+        # options from subclass
+        self._filter_data()
+        
+    def _filter_data(self):
+        raise NotImplementedError
+        
+    def _get_data(self):
+        raise NotImplementedError
+        
+    def _cache_data(self):
+        # dir for saving file
+        basedir = f'{self.config["OutputPath"]}/{self.config["RunName"]}'
+        write_dir = f'{basedir}/{self.myname}_plumes'
+        if not os.path.exists(write_dir):
+            os.makedirs(write_dir)
+        # write file to geojson
+        self.gdf.to_file(
+            f'{write_dir}/plumes.geojson',
+            driver = 'GeoJSON' 
+        )
+
+
+class SRON(PlumeObserver):
+    '''
+    SRON point source data retrieved
+    from the API. Converts data to GeoDataFrame
+    with point geometries.
+
+    Parameters
+    ----------
+    config: dict of IMI config.yml contents
+    usecached: bool, whether to use saved data
+    cache: None or path to existing file. Will search 
+        for saved data in IMI output dir if no
+        cache given.
+
+    Use
+    ---
+    sron = SRON(config, cache=cache_file)
+
+    Access the GeoDataFrame with data:
+    
+    sron.gdf
+    
+
+    '''
+    def __init__(self, config, usecached = True, cache = None):
+        #if ((not usecached) | (cache is None)):
+        #    msg = 'Must supply cached file for SRON plumes'
+        #    sys.exit(msg)
+        self.myname = 'SRON'
+        #self.config = config
+        self.cache = cache
+        super().__init__(self.myname, config, usecached, self.cache)
+        
+    def _filter_data(self):
+        pass
+
+    def _get_data(self):
+        
+        # get data
+        print(f"Fetching plumes from {self.myname}...")
+        
+        # URL of the SRON database for weekly methane plumes
+        sron_url = 'https://ftp.sron.nl/pub/memo/CSVs/'
+        
+        response = requests.get(sron_url)
+        if not response.status_code == 200:
+            msg = (
+                f"Couldn't access SRON site {sron_url}.  "
+                f"failed with error {response.status_code}: "
+                f"{response.reason}.\n"
+                "SRON plumes not retrieved.  "
+                "Check SRON website or remove SRON from "
+                "point source list in config file."
+            )
+            raise Exception(msg)
+        
+        parser = BeautifulSoup(response.content, "html.parser")
+        
+        dfplume = pd.DataFrame()
+        
+        # retrieve csvs, read into dataframe
+        dfplume = pd.DataFrame()
+        for link in parser.find_all("a"):
```

Comment:
If you wanted to be fancy and speed up downloads you could try either parallelizing this or using multiple threads. Probably not a priority though unless it is very slow

Reply:
It's not too slow, I'll leave it as is unless you disagree